packages/indexer-cli/src/command-helpers.ts

```diff
@@ -206,3 +207,47 @@ export function suggestCommands(
   )
   return suggestions.length > 0 ? suggestions : supported_commands
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function extractProtocolNetworkOption(
+  options: {
+    [key: string]: any
+  },
+  required = false,
+): string | undefined {
+  const { n, network } = options
+
+  // Tries to extract the --network option from Gluegun options.
+  // Throws if required is set to true and the option is not found.
+  if (!n && !network) {
+    if (required) {
+      throw new Error("The option '--network' is required")
+    } else {
+      return undefined
+    }
+  }
+
+  // Check for invalid usage
+  const allowedUsages =
+    (n === undefined && typeof network === 'string') ||
+    (network === undefined && typeof n === 'string')
+  if (!allowedUsages) {
+    throw new Error("Invalid usage of the option '--network'")
+  }
+  const input = (network ?? n) as string
+
+  try {
+    return validateNetworkIdentifier(input)
+  } catch (parseError) {
+    throw new Error(`Invalid value for the option '--network'. ${parseError}`)
+  }
+}
+
+// Same as `extractProtocolNetworkOption`, but always require the --network option to be set
+export function requireProtocolNetworkOption(options: { [key: string]: any }): string {
```

Comment:
Worth having separate functions? Couldn't one just use extractProtocolNetworkOption with required=true?

Reply:
While it might appear redundant, `extractProtocolNetwork` has a nullable return type, whereas `requireProtocolNetwork` has a stricter return type. This difference significantly aids the code where they are used, as it eliminates the need for conditional checks to account for missing values.